### DISCUSSION TITLE
zsh: command not found: axmol

### DISCUSSION THREAD

**Discussion by stefan80** [axmolengine/axmol#1749] [General]
Hi,
I have updated to latest axmol build and want to create a project but axmol is not found.
My setup is a Mac. Powershell is installed, pwsh setup.ps1 was successfully executed. But afterwards, the axmol command cannot be executed to create a project.
Do I have to setup environment variables?

**Comment by rh101** [comment]
Just out of curiosity, did you restart the terminal instance?

Check the `[HOME_DIR]/.zhshrc` file, as the setup process should have added something like these lines:

```
# Add environment variable AX_ROOT for axmol
export AX_ROOT="/Path/To/axmol"
# Add axmol console tool to PATH
export PATH=$AX_ROOT/tools/console:$PATH
```

Also, are you running the axmol command inside Powershell? If not, just run `pwsh` on its own, and inside that run all the axmol commands.

**Comment by stefan80** [comment]
@rh101 You guided me to the right direction. I opened the zhshrc file and recognized that the entries for Axmol were missing. I added them but still had issues with the setup.ps1 script of axmol 1.0 release. The latest release is working fine. Since I do not want to upgrade my project to 2.1.2, I switched to an older release ~1.0 with python script and everything runs well now.
Thank you very much!

By the way, do you know, if it is possible to upgrade an existing project to the latest 2.1.2 by use of the axmol tool?

**Reply by rh101** [reply]
I'm not really sure if the migrate tool works for Axmol 1.0 to Axmol 2.0 project conversion, or if there is another tool available, but @halx99 would know.

Other than that, I strongly suggest that you move to the latest version of Axmol, because it is better than 1.0.

The quickest way is to do this:

1 - Backup your Axmol 1.0 project
2 - Create a new Axmol 2.x project with the same name as your existing project
3 - Merge in the files and any changes you require to `CMakeLists.txt`/`proj.android` etc. etc.

The biggest difference will be if you have custom shaders.  Moving the shaders to the new format isn't complex, and for examples, check these posts for original [v1.0 shaders](https://github.com/axmolengine/axmol/discussions/1660#discussioncomment-8416427), and updated [v2.x shaders](https://github.com/axmolengine/axmol/discussions/1660#discussioncomment-8416574).

The custom shaders should be located in `YourProject/Source/shaders/`, and the build process will convert them to the correct format required by the build target platform.

If you have any issues with migrating, then just ask for help.

**Reply by stefan80** [reply]
Hi @rh101 , thanks a lot for your help and advice. I really appreciate that! I was now able to build a 1.0 and a 2.1.2 project on my macbook. The 1.0 project runs as expected but the 2.1.2 does not offer anymore the ZipFile method createWithBuffer
`ZipFile* lZipFile = ZipFile::createWithBuffer(lZip->data, lZip->size);`
I load my big txt assets from zipped data structures and so I am not sure how to proceed there at the moment. I saw that @halx99 did a fix in the past for cocos2d-x there [](https://github.com/cocos2d/cocos2d-x/issues/20182) but did not figure out yet if that could help.

**Reply by rh101** [reply]
I can see `ZipFile::createWithBuffer` it was removed in commit https://github.com/axmolengine/axmol/commit/79187cf9f61e00516f04ee1bc9fed7ce14f8676a but I'm unsure why.  @halx99 is it possible to add that same functionality back in, to be able to load from a memory buffer rather than a file, or is there a technical reason why it can't be done?
